config/config.go

```diff
@@ -0,0 +1,100 @@
+    ZookeeperURL                   string      `env:"KAFKA_ZOOKEEPER_ADDR"                          flag:"zookeeper-addr"                               flagDesc:"Main CH Zookeeper address"`
+    RetryThrottleRate              int         `env:"RETRY_THROTTLE_RATE_SECONDS"                   flag:"retry-throttle-rate-seconds"                  flagDesc:"Retry throttle rate seconds"`
+    MaxRetryAttempts               int         `env:"MAXIMUM_RETRY_ATTEMPTS"                        flag:"max-retry-attemps"                            flagDesc:"Maximum retry attempts"`
+    IsErrorConsumer                bool        `env:"IS_ERROR_QUEUE_CONSUMER"                       flag:"is-error-queue-consumer"                      flagDesc:"Set this flag if it is an error queue consumer"`
+    ChsAPIKey                      string      `env:"CHS_API_KEY"                                   flag:"chs-api-key"                                  flagDesc:"API access key"`
+    SchemaRegistryURL              string      `env:"SCHEMA_REGISTRY_URL"                           flag:"schema-registry-url"                          flagDesc:"Schema registry url"`
+    PaymentsAPIURL                 string      `env:"PAYMENTS_API_URL"                              flag:"payments-api-url"                             flagDesc:"Base URL for the Payment Service API"`
+    MongoDBURL                     string      `env:"MONGODB_URL"                                   flag:"mongodb-url"                                  flagDesc:"MongoDB server URL"`
+    Database                       string      `env:"RECONCILIATION_MONGODB_DATABASE"               flag:"mongodb-database"                             flagDesc:"MongoDB database for data"`
+    TransactionsCollection         string      `env:"MONGODB_PAYMENT_REC_TRANSACTIONS_COLLECTION"   flag:"mongodb-payment-rec-transactions-collection"  flagDesc:"MongoDB collection for payment transactions data"`
+    ProductsCollection             string      `env:"MONGODB_PAYMENT_REC_PRODUCTS_COLLECTION"       flag:"mongodb-payment-rec-products-collection"      flagDesc:"MongoDB collection for payment products data"`
+}
+
+// ProductMap contains a map of product codes
+type ProductMap struct {
+    Codes map[string]int `yaml:"product_code"`
+}
+
+// ServiceConfig returns a ServiceConfig interface for Config
+func (c Config) ServiceConfig() ServiceConfig {
+    return ServiceConfig{c}
+}
+
+// ServiceConfig wraps Config to implement service.Config
+type ServiceConfig struct {
+    Config
+}
+
+// Namespace implements service.Config.Namespace
+func (c *Config) Namespace() string {
+    return "payment-reconciliation-consumer"
+}
+
+var productMap *ProductMap
+
+// GetProductMap fetches a map of product codes
+func GetProductMap() (*ProductMap, error) {
+
+    if productMap != nil {
+        return productMap, nil
+    }
+
+    filename, err := filepath.Abs("config/productCode.yml")
```

Comment:
I had issues when reading a yml file in lfp-pay-api. Here's the PR that fixed it for us: https://github.com/companieshouse/lfp-pay-api/pull/14/files

Basically, move the yaml file to an assets folder and copy it over to tmpdir in the make file

Reply:
Thanks for that. The yaml file has been moved and the `Makefile` updated